File: pkg/models/event.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Event struct {
	ID                   uuid.UUID `gorm:"primaryKey"`
	OwnerID              uuid.UUID
	Title                string `json:"email" gorm:"unique; not null"`
	Description          string `json:"password" gorm:"not noll"`
	NumberOfParticipants int
	Location             string
	Private              bool `gorm:"default:false"`
	StartDate            time.Time
	EndDate              time.Time
	DateCreated          time.Time
	DateUpdated          time.Time `gorm:"autoCreateTime"`

	User  User   `gorm:"foreignKey:OwnerID;unique"`
	Users []User `gorm:"many2many:participants;"`

	CategoryID uuid.UUID `gorm:"unique"`
	Category   Category  `gorm:"foreignKey:CategoryID"`
}

File: src/db/schema/Event.ts
import {
  pgTable,
  varchar,
  timestamp,
  integer,
  date,
  boolean
} from 'drizzle-orm/pg-core'

import { createId } from '@paralleldrive/cuid2'
import { users } from './Users'
import { category } from './Category'

export const event = pgTable('event', {
  id: varchar('id')
    .$defaultFn(() => createId())
    .primaryKey(),

  title: varchar('title').notNull(),
  description: varchar('description').notNull(),

  participants: integer().default(0).notNull(),
  maxParticipants: integer(),

  isOffline: boolean().default(true),
  location: varchar('location'),
  url: varchar('url'),

  startDate: date().notNull(),
  endDate: date(),

  dateCreated: timestamp({ withTimezone: true }).defaultNow(),
  dateUpdated: timestamp({ withTimezone: true }).defaultNow(),

  ownerId: varchar('owner_id')
    .notNull()
    .references(() => users.id),
  categoryId: varchar('category_id')
    .notNull()
    .references(() => category.id)
})

export const eventTable = {
  event
} as const

export type EventTable = typeof eventTable
